core/src/replay_stage.rs

```diff
@@ -1727,20 +1727,27 @@ impl ReplayStage {
         // All errors must lead to marking the slot as dead, otherwise,
         // the `check_slot_agrees_with_cluster()` called by `replay_active_banks()`
         // will break!
-        blockstore_processor::confirm_slot(
-            blockstore,
-            bank,
-            &mut w_replay_stats,
-            &mut w_replay_progress,
-            false,
-            transaction_status_sender,
-            Some(replay_vote_sender),
-            None,
-            verify_recyclers,
-            false,
-            log_messages_bytes_limit,
-            prioritization_fee_cache,
-        )?;
+
+        let mut more_entries_to_process = true;
+        // more entries may have been received while replaying this slot.
+        // looping over this ensures that slots will be processed as fast as possible with the
+        // lowest latency.
+        while more_entries_to_process {
```

Comment:
I think we might want some kind of exit out of here based on timing as a safe-guard. Otherwise we are potentially not voting for a long time or starting a leader slot which wouldn't be great.

@carllin wdyt?

Reply:
i agree that'd be the safest thing to do, could limit to 400-ish ms. worst case exec takes longer than expected and we just hit another iteration of replay before hitting this again

Reply:
yeah this isn't safe as is because leaders often abandon minor forks and don't notify the network until their next block. Malicious leaders may never notify the network, so you'll never see the slot full, and you'll spin in this loop for a long time/forever.

The sleep also seems hacky. A better fix would be to change the blockstore signaling to notify `ReplayStage` of slots  that have new entries ready for replay, and consume that signal insterad.

Reply:
Sleep may not be necessary, worst case we just hit one more iteration of reading from blockstore before returning here right: https://github.com/solana-labs/solana/blob/0145447d60ee3ced39d1deb7735c3b8c0e5b5da4/ledger/src/blockstore_processor.rs#L1151-L1153

Reply:
i don't think sleep is the right word, more like:
```
let start = Instant::now();
while more_entries_to_process && start.elapsed() < Duration::from_millis(TIMEOUT) {
    confirm_slot(...);
}
```

Reply:
hmmm yeah that works, should prevent bad leader from DOSing by continually streaming new shreds

Reply:
Could even probably warn if it hits that timeout